### PR TITLE
support keeping aliases up to date

### PR DESCRIPTION
**Pull request by azylman**
This adds an `/aliases/update` endpoint that updates all specified time-based aliases based on the config. It also adds an optional cron time to update the aliases at.

### CODE REVIEW

File: lib/elasticsearch.ts [hunk 1]
```diff
@@ -70,14 +78,108 @@ function filter_old_indices(current_indices) {
...
-  return Promise.all(_.map(indices, (index) => delete_index(index)));
+  return Promise.all(_.map(indices, delete_index));
 }
 
 export function clear_old_indices() {
   return get_indices().then(filter_old_indices).then(delete_indices);
 }
+
+function get_aliases() {
+  return new Promise((resolve, reject) => {
```
Comment (burnsed) [inline lib/elasticsearch.ts]: do you need a new Promise here? I thought that returning the `then()` would chain this to the promise returned from `get_es` ?

File: lib/elasticsearch.ts [hunk 2]
```diff
@@ -70,14 +78,108 @@ function filter_old_indices(current_indices) {
...
+    const action_el = (action, index) => {
+      const out = {};
+      out[action] = {index, alias};
+      return out;
+    };
+    const actions = _.map(indices_to_remove, _.partial(action_el, "remove"))
+      .concat(_.map(indices_to_add, _.partial(action_el, "add")));
+    request_es("post", "/_aliases", {actions}).then(() => {
+      resolve();
+    }).catch((err) => {
```
Comment (burnsed) [inline lib/elasticsearch.ts]: these two blocks could probably use some comments to explain the format needed for posting to `/_aliases`

File: lib/elasticsearch.ts [hunk 3]
```diff
@@ -70,14 +78,108 @@ function filter_old_indices(current_indices) {
...
+      }
+
+      resolve(aliases);
+    }).catch((err) => {
+      reject(err);
+    });
+  });
+}
+
+function remove_unmanaged_aliases(aliases) {
```
Comment (burnsed) [inline lib/elasticsearch.ts]: NIT: filter might be better than "remove" in the name here, or a brief comment/doc string to describe what it does would also work.

**Comment by burnsed** [comment]
a few nits, LGTM

File: lib/elasticsearch.ts (hunk 1 above)
Comment (azylman) [inline lib/elasticsearch.ts]: This is my first time using promises, the more I learn the more I like them: https://github.com/Clever/elasticsearch-toolbox/pull/4/commits/6792ab48afe00a85a94f6177b0d7f0ad728c2243

File: lib/elasticsearch.ts (hunk 3 above)
Comment (azylman) [inline lib/elasticsearch.ts]: https://github.com/Clever/elasticsearch-toolbox/pull/4/commits/4d9af1f09942daf75a11604da019f4a3fede8e74

File: lib/elasticsearch.ts (hunk 2 above)
Comment (azylman) [inline lib/elasticsearch.ts]: https://github.com/Clever/elasticsearch-toolbox/pull/4/commits/1881e50cf944a6678edd711519453853df45d39e

**Pull request merged**